iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/helpers/rules/BasicTierHubOnlyTestRule.java

```diff
@@ -0,0 +1,40 @@
+package com.microsoft.azure.sdk.iot.common.helpers.rules;
+
+import com.microsoft.azure.sdk.iot.common.helpers.IntegrationTest;
+import com.microsoft.azure.sdk.iot.common.helpers.annotations.BasicTierHubOnlyTest;
+import com.microsoft.azure.sdk.iot.common.helpers.annotations.IotHubTest;
+import org.junit.Assume;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class BasicTierHubOnlyTestRule implements TestRule
+{
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new IgnorableStatement(base, description);
+    }
+
+    private class IgnorableStatement extends Statement {
+
+        private final Statement base;
+        private final Description description;
+
+        public IgnorableStatement(Statement base, Description description) {
+            this.base = base;
+            this.description = description;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
```

Comment:
This method is called prior to running any test method, and can be used to determine if the test should be run or skipped at runtime